jax_triton/pallas/pallas_call.py

```diff
@@ -221,42 +269,110 @@ def clear_caches():
   _initial_style_open_jaxpr.cache_clear()
   _compile_jaxpr.cache_clear()
 
-def pallas_call(f, out_shape, grid, *, debug: bool = False,
+def _preprocess_grid(grid: Optional[Union[Grid, int]]) -> Grid:
+  if grid is None:
+    return (1,)
+  if isinstance(grid, int):
+    return (grid,)
+  return grid
+
+def _extract_function_name(f: Callable, name: Optional[str]) -> str:
+  if name is None:
+    name = f.__name__ if hasattr(f, "__name__") and f.__name__ else "func"
+  return name
+
+def _convert_block_spec_to_block_mapping(
+    grid: Grid, block_spec: Optional[BlockSpec]) -> Optional[BlockMapping]:
+  if block_spec is None:
+    return None
+  in_avals = [jax_core.ShapedArray((), jnp.int32) for _ in grid]
+  jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(
+      lu.wrap_init(block_spec.index_map), in_avals)
+  if consts:
+    raise NotImplementedError
```

Comment:
What's wrong with consts? You're bundling them into a `ClosedJaxpr` correctly

Reply:
Fixed.